### ISSUE TITLE
suggestion: add native support for create index against edge in/out properties

### ISSUE THREAD

**Issue by danrub** [orientechnologies/orientdb#3727]
You should allow create index against edge in/out (build in) properties without forcing creating them manually on the edge it self.

Today when I want to create index against this properties, the db will not allow me to do that, only after re-definition of the related class/base class schema. 
This workaround causing the server to log warning ("Found property in/out against E") on each database open. this may cause major performance penalty, if server running inside multi user demanding environment (>500 Tx/s ).
Also, the log is filled with unnecessary warnings, filling up disk space.

**Comment by martingg88** [comment]
any update?

**Comment by tglman** [comment]
hi,

We have still a few things to improve in the schema management for allow to create the in/out properties by default, for example we cannot add a constraint only on a subclass:
```
create property E.in (LINK);
create property E.out (LINK);

create class Fried extends E;
alter property Friend.in linkedclass Person;// I cannot do this today
alter property Friend.out linkedclass Person;// I cannot do this today
```

So before we add this we should improve the schema management on this parts.

Regards

**Comment by zeeshan2709** [comment]
@tglman: are you adding this feature in the upcoming release?

**Comment by tglman** [comment]
Hi,

This is not done in 3.0-M1 and won't be done for 3.0-M2, it may be included in the 3.0-Final but is not yet sure.

Regards
